common/src/main/java/io/homeassistant/companion/android/common/data/websocket/impl/WebSocketRepositoryImpl.kt

```diff
@@ -64,11 +74,11 @@ class WebSocketRepositoryImpl @Inject constructor(
     private var connection: WebSocket? = null
     private val connectedMutex = Mutex()
     private var connected = Job()
-    private val stateChangedMutex = Mutex()
-    private var stateChangedFlow: SharedFlow<StateChangedEvent>? = null
+    private val eventSubscriptionMutex = mutableMapOf<String, Mutex>()
```

Comment:
The more I'm thinking about this, we should just use a single Mutex rather than one per event type.  Technically this could cause a race condition the way it's written.

Reply:
I didn't really think much about it, but I guess it could if you try to subscribe to the same event again within a few ms. Should be a fairly quick message so any delay because you're now waiting for other subscribe_event calls to complete is negligible.